include/systems/fem_context.h

```diff
@@ -776,18 +842,22 @@ class FEMContext : public DiffContext
 
   /**
    * Finite element objects for each variable's interior, sides and edges.
+   * We store FE objects for each element dimension present in the mesh,
+   * except for edge_fe which only applies to 3D elements.
    */
-  std::map<FEType, FEAbstract*> _element_fe;
-  std::map<FEType, FEAbstract*> _side_fe;
+  std::map<unsigned char, std::map<FEType, FEAbstract*> > _element_fe;
+  std::map<unsigned char, std::map<FEType, FEAbstract*> > _side_fe;
   std::map<FEType, FEAbstract*> _edge_fe;
 
 
   /**
    * Pointers to the same finite element objects, but indexed
-   * by variable number
+   * by variable number. We store FE objects for each element dimension
+   * present in the mesh, except for edge_fe_var which only applies
+   * for 3D elements.
    */
-  std::vector<FEAbstract*> _element_fe_var;
-  std::vector<FEAbstract*> _side_fe_var;
+  std::map<unsigned char, std::vector<FEAbstract*> > _element_fe_var;
+  std::map<unsigned char, std::vector<FEAbstract*> > _side_fe_var;
   std::vector<FEAbstract*> _edge_fe_var;
```

Comment:
OK, there actually is a comment there, it's just not showing up on the diff because the diff is ugly.